File: Part1_강의자료/Part1_연습문제_풀이/1339_단어수학/1339.cpp
#include <stdio.h>
#include <string.h>
#include <algorithm>
using namespace std;

int main()
{
    int N;
    int alp_value[26] = {0, };
    char word[11];

    scanf("%d", &N);
    while (N--)
    {
        scanf("%s", word);
        int place_value = 1;
        for (int i = strlen(word) - 1; i >= 0; i--)
        {
            alp_value[word[i] - 'A'] += place_value;
            place_value *= 10;
        }
    }

    sort(alp_value, alp_value + 26);

    int ans = 0;
    for (int i = 0; i < 10; i++)
        ans += alp_value[25 - i] * (9 - i);
    printf("%d\n", ans);
    return 0;
}

File: Part1_강의자료/Part1_연습문제_풀이/1339_단어수학/1339.py
N = int(input())
alp_value = [0] * 26

for _ in range(N):
    word = input()
    place_value = 1
    for i in range(len(word)-1, -1, -1):
        alp_value[ord(word[i]) - ord('A')] += place_value
        place_value *= 10

alp_value = sorted(alp_value)

ans = 0
for i in range(10):
    ans += alp_value[25 - i] * (9 - i)

print(ans)
